web/html/src/manager/systems/all-list.tsx

```diff
@@ -0,0 +1,170 @@
+
+    setSelectedSystems(items);
+  };
+
+  return (
+    <>
+      <h1>
+        <IconTag type="header-system" />
+        {t(" Systems ")}
+        <a
+          href={`/docs/${props.docsLocale}/reference/systems/systems-list.html`}
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          <IconTag type="header-help" />
+        </a>
+      </h1>
+
+      <Table
+        data="/rhn/manager/api/systems/list/all"
+        identifier={(item) => item.id}
+        initialSortColumnKey="server_name"
+        selectable={(item) => item.hasOwnProperty("id")}
+        selectedItems={selectedSystems}
+        onSelect={handleSelectedSystems}
+        searchField={<SearchField options={allListOptions} />}
+        defaultSearchField="server_name"
+        emptyText={t("No Systems.")}
+      >
+        <Column
+          columnKey="server_name"
+          comparator={Utils.sortByText}
+          header={t("System")}
+          cell={(item) => Systems.iconAndName(item)}
+        />
+        <Column
+          columnKey="statusType"
+          comparator={Utils.sortByText}
+          header={t("Updates")}
+          cell={(item) => {
+            if (item.statusType == null) {
+              return "";
+            }
+            return Systems.statusDisplay(item, props.isAdmin);
+          }}
+        />
+        <Column
+          columnKey="totalErrataCount"
+          comparator={Utils.sortByText}
+          header={t("Patches")}
+          cell={(item) => {
+            let totalErrataCount = item.securityErrata + item.bugErrata + item.enhancementErrata;
+            if (totalErrataCount !== 0) {
+              return <a href={`/rhn/systems/details/ErrataList.do?sid=${item.id}`}>{totalErrataCount}</a>;
+            }
+            return totalErrataCount;
+          }}
+        />
+
+        <Column
+          columnKey="outdatedPackages"
+          comparator={Utils.sortByText}
+          header={t("Packages")}
+          cell={(item) => {
+            if (item.outdatedPackages !== 0) {
+              return (
+                <a href={`/rhn/systems/details/packages/UpgradableList.do?sid=${item.id}`}>{item.outdatedPackages}</a>
+              );
+            }
+            return item.outdatedPackages;
+          }}
+        />
+
+        <Column
+          columnKey="extraPkgCount"
+          comparator={Utils.sortByText}
+          header={t("Extra Packages")}
+          cell={(item) => {
+            if (item.extraPkgCount !== 0) {
+              return (
+                <a href={`/rhn/systems/details/packages/ExtraPackagesList.do?sid=${item.id}`}>{item.extraPkgCount}</a>
+              );
+            }
+            return item.outdatedPackages;
+          }}
+        />
+
+        <Column
+          columnKey="configFilesWithDifferences"
+          comparator={Utils.sortByText}
+          header={t("Config Diffs")}
+          cell={(item) => {
+            if (item.configFilesWithDifferences !== (0 || null)) {
```

Comment:
This conditional probably does not do what you think it does. `0 || null` always evaluates to `null`, so the condition is equivalent to `if (item.configFilesWithDifferences !== null) {}`. If you want to check for either `0` or `null`, you could check them separately, or alternatively if `undefined` is not a state you need to handle separately, you could do `if (item.configFilesWithDifferences) {}`

Reply:
I believe we can just test again non `0` here.